### ISSUE TITLE
[feature request] switch php version for apache

### ISSUE THREAD

**Issue by jhdxr** [phpbrew/phpbrew#708]
add the ability to switch php version for apache, like what `phpbrew install` does.

**Comment by c9s** [comment]
This is also related to:

https://github.com/phpbrew/phpbrew/issues/339
https://github.com/phpbrew/phpbrew/issues/587

**Comment by c9s** [comment]
I am thinking the apache switching could be a flag of  "switch"

**Comment by jhdxr** [comment]
Yes, it's an alternative I've thought of. I think we should provide the ability to switch version for terminal(cli+fpm) / apache / both. 
so maybe
`phpbrew switch` terminal only 
`phpbrew switch --for-apache` apache only
`phpbrew switch --with-apache` both

**Comment by c9s** [comment]
Sounds good

**Comment by jhdxr** [comment]
Or we can separate the apache relating things to a new command, for example
`phpbrew apache switch` switch php version
`phpbrew apache generate-config` generate/regenerate config file
`phpbrew apache start / restart / stop` start / restart / stop apache server

however I'm wondering should we add the start/stop part, because it's more relevant to web server rather than php itself.

**Comment by c9s** [comment]
This is even better

xiami <notifications@github.com> 於 2016年4月28日 星期四寫道：

**Comment by c9s** [comment]
But I think `apache2 switch` should already include `apache generate-config`, the `phpbrew apache generate-config` can be removed I guess.

**Comment by c9s** [comment]
Another related issue https://github.com/phpbrew/phpbrew/issues/226#issuecomment-170124212

**Comment by jdrsqrd** [comment]
Until this feature is added how can I manually switch the version of php that apache uses?

**Comment by jhdxr** [comment]
@jdrsqrd edit your apache config file, edit the `LoadModule phpXXXXX` lines to the library you'd like to use, then restart your apache

**Comment by jdrsqrd** [comment]
@jhdxr 
Sorry I'm a little confused. Do I need the .so files to point to in `/etc/apache2/mods-available/php5.load`? or should I just need to only change my `/etc/apache2/apache.conf` file? (I tried the latter and got an error on restarting apache, and I can't find the proper .so file for the php5.load file)

**Comment by jhdxr** [comment]
You need to edit the `/etc/apache2/apache.conf` only, and you should specify the fullpath, lilke `LoadMoudle php5_module /usr/lib64/http/modules/lib5.5.34.so`. these so file were generated when you built php with --apxs2

**Comment by c9s** [comment]
Hi all,

I've created 2.0 branch to start the development of more new features that have breaking changes.
Please rebase your branch with the 2.0 branch to start the development.

cc @marcioAlmada @shinnya

**Comment by thomashery** [comment]
Hi @c9s ,
are you still considering this feature? I love phpbrew seriously great project. 
However, considering that most users would use PHP with apache, having to follow so many manual steps to ACTUALLY switch versions of Php does defeat the purpose of phprew...
I suggest adding support to apache out of the box since at the moment all the steps to follow to install and switch to a version of php are time consuming: Installing php versions for apache with aspx2, using the right extensions when building, and changing the apache config to load the right version of php, etc...

**Comment by 3kynox** [comment]
I agree, found this place while searching how to activate phpbrew switching in apache ...

**Comment by roland-d** [comment]
We are 3 years on and I am also looking for a script like this. I did find something on StackOverflow at https://stackoverflow.com/a/34686059

I am/was using brew with this `sphp` script https://gist.github.com/rhukster/f4c04f1bf59e0b74e335ee5d186a98e2/

Perhaps it can serve as some inspiration.
